iceberg/src/main/scala/org/apache/spark/sql/delta/icebergShaded/IcebergTransactionUtils.scala

```diff
@@ -80,27 +84,49 @@ object IcebergTransactionUtils
         // throw an exception when building the data file.
         .withRecordCount(add.numLogicalRecords.getOrElse(-1L))
 
-    if (add.stats != null && add.stats.nonEmpty) {
```

Comment:
Given now we are using the new `convertDeltaAddFileToIcebergDataFile(..., schema)` to do all `append, rewrite, overwrite` conversions, this `convertDeltaAddFileToIcebergDataFile(.., deltaLog)` is only used in tests (`IcebergStatsConversionSuite`). Should we remove it and modify according tests?

Reply:
It's good question, I was thinking about this but unfortunately the existing `convertDeltaAddFileToIcebergDataFile` is actually under the public API section and I don't think we want to remove that without going through some deprecation process? Some users could be leveraging it technically. I don't know what the process for deprecation is in Delta. But I would recommend we take that in a follow on.

Once the deprecation is done, then we can also update the IcebergStatsConversionSuite but until then we may want to keep those tests since those APIs are exposed and can be used by other. We have round-trip coverage of that path through the new insert partition tests.

Reply:
Makes sense!

Reply:
Discussed with @lzlfred this API was never intended to be made public and it *should* be safe/reasonable to simplify this so we just have 1. So updated it @ChengJi-db !